File: tests/conformance/fixtures/typescript/15-diamond-deps.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('A').run('echo A');
p.task('B').run('echo B').after('A');
p.task('C').run('echo C').after('A');
p.task('D').run('echo D').after('B', 'C');

p.emit();

File: tests/conformance/fixtures/python/15-diamond-deps.py
from sykli import Pipeline

p = Pipeline()

p.task("A").run("echo A")
p.task("B").run("echo B").after("A")
p.task("C").run("echo C").after("A")
p.task("D").run("echo D").after("B", "C")

p.emit()
